packages/core/core/src/RequestTracker.js

```diff
@@ -354,104 +374,64 @@ export default class RequestTracker {
   ) {
     this.graph.replaceSubrequests(requestId, subrequestNodes);
   }
-}
 
-type RequestRunnerOpts = {
-  tracker: RequestTracker,
-  ...
-};
+  async runRequest<TInput, TResult, TRequest: Request<TInput, TResult>>(
+    request: TRequest,
+  ): Async<TResult> {
+    let id = request.id;
 
-export type RunRequestOpts = {|
-  signal?: ?AbortSignal,
-  parentId?: string,
-|};
-
-export type RequestRunnerAPI = {|
-  invalidateOnFileCreate: Glob => void,
-  invalidateOnFileDelete: FilePath => void,
-  invalidateOnFileUpdate: FilePath => void,
-  invalidateOnStartup: () => void,
-  replaceSubrequests: (Array<RequestGraphNode>) => void,
-  storeResult: (result: mixed) => void,
-  getId: () => string,
-|};
-
-export function generateRequestId(type: string, request: mixed) {
-  return md5FromObject({type, request});
-}
-
-export class RequestRunner<TRequest, TResult> {
-  type: string;
-  tracker: RequestTracker;
-
-  constructor({tracker}: RequestRunnerOpts) {
-    this.tracker = tracker;
-  }
+    if (this.hasValidResult(id)) {
+      // $FlowFixMe
+      let result: TResult = (this.getRequestResult(id): any);
```

Comment:
make it `getRequestResult<T>(id): T` and then `this.getRequestResult<TResult>(id)`?

That would move the casting to any into `getRequestResult`

Reply:
Thanks for the help @mischnic :)